src/server_manager/model/gcp.ts

```diff
@@ -12,9 +12,53 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// TODO: This is just a stub atm and will need to define and implement the rest
-//  of the functionality.
+import {ManagedServer, RegionId} from './server';
+
+// Keys are region IDs like "us-central1".
+// Values are zones like ["us-central1-a", "us-central1-b"].
+// TODO: Create separate type
+export type RegionMap = {
+  [regionId: string]: RegionId[]
+};
+
+/**
+ * The Google Cloud Platform account model.
+ */
 export interface Account {
-  // Returns a user-friendly name associated with the account.
+  /**
+   * Returns a globally unique identifier for this Account.
+   */
+  getId(): string;
+
+  /**
+   * Returns a user-friendly name associated with the account.
+   */
   getName(): Promise<string>;
+
+  /**
+   * Creates an Outline server on a Google Compute Engine VM instance.
+   *
+   * This method returns after the VM instance has been created. The Shadowbox
+   * Outline server may not be fully installed. See {@link ManagedServer#waitOnInstall}
+   * to be notified when the server installation has completed.
+   *
+   * @param projectId - The GCP project ID.
+   * @param name - The name to be given to the server.
+   * @param zoneId - The ID of the GCP zone to create the server in.
+   */
+  createServer(projectId: string, name: string, zoneId: string): Promise<ManagedServer>;
+
+  /**
+   * Lists the Outline servers in a given GCP project.
+   *
+   * @param projectId - The GCP project ID.
+   */
+  listServers(projectId: string): Promise<ManagedServer[]>;
+
+  /**
+   * Lists the Google Compute Engine locations available to given GCP project.
+   *
+   * @param projectId - The GCP project ID.
+   */
+  listLocations(projectId: string): Promise<Readonly<RegionMap>>;
```

Comment:
We can't handle regions from different clouds the same way. This will break when you do the UI.
We should encode the fact that regions in GCP are different from regions in DO. You should use a GCP-specific `ZoneId` instead, since the code calling this must be GCP-aware. You can even call the method `listZones()`.

Alternatively, you could have a generic `CloudLocation` type that can return an id, the country code (for flag)  and city id (for name)

Reply:
Yep, I agree. I actually copied and redefined RegionMap in this file with the intention to make it GCP specific.

You can see a bit of my thought process in the comments above.

Reply:
The RegionMap is good, but can you rename to ZoneMap?
Also define a ZoneId type there and stop depending on RegionId.